File: py/pe20.py
#! /usr/bin/env python3
 
from sys import argv
from pe import fact, sod

def pe20(n):
    """
    Find the sum of digits in 100!
    """
    print("sum of digits in %d! is %d" % (n, sod(fact(n))))

def main():
    argc = len(argv)
    if argc > 1:
        try:
            n = int(argv[1])
        except ValueError:
            print("  python pe20.py [integer]")
            return
    else:
        n = 100
    
    pe20(n)

if __name__ == "__main__":
    main()

File: cpp/cli/src/pe20.cc
#include "../include/pe20.h"

#include <cstdlib>
#include <iostream>

int pe20(int argc, char **argv)
{
  using namespace std;

  uint32_t n = 100;
  if (argc > 1) {
    char *end;
    n = strtoul(argv[1], &end, 10);
  }

  cout << "n = " << n << endl;
  cout << "sum of digits = " << pe::pe20(n) << endl;

  return 0;
}
